src/storage/src/storage_state/async_storage_worker.rs

```diff
@@ -66,21 +67,29 @@ pub enum AsyncStorageWorkerResponse<T: Timestamp + Lattice + Codec64> {
         GlobalId,
         IngestionDescription<CollectionMetadata>,
         Antichain<T>,
-        Vec<Row>,
+        BTreeMap<GlobalId, Vec<Row>>,
     ),
 }
 
 async fn reclock_resume_frontier<C, IntoTime>(
     persist_clients: &PersistClientCache,
     ingestion_description: &IngestionDescription<CollectionMetadata>,
-    resume_upper: &Antichain<IntoTime>,
-) -> Antichain<C::Time>
+    resume_uppers: &BTreeMap<GlobalId, Antichain<IntoTime>>,
+) -> BTreeMap<GlobalId, Antichain<C::Time>>
 where
     C: SourceConnection + SourceRender,
     IntoTime: Timestamp + Lattice + Codec64 + Display,
 {
-    if **resume_upper == [IntoTime::minimum()] {
-        return Antichain::from_elem(C::Time::minimum());
+    let mut reclocked_resume_frontiers = BTreeMap::new();
+
+    for (id, resume_upper) in resume_uppers {
+        if **resume_upper == [IntoTime::minimum()] {
+            reclocked_resume_frontiers.insert(*id, Antichain::from_elem(C::Time::minimum()));
```

Comment:
We chatted about this and we needed something to handle the case where the remap shard hasn't been written to yet, so cannot be subscribed; we agreed to instead use the write handle upper because that also gives us the ability to make further assertions about the state of the resume uppers.